### ISSUE TITLE
[Bug] zos_job_submit truncates final character of input

### ISSUE THREAD

**Issue by ind1go** [ansible-collections/ibm_zos_core#815]
### Is there an existing issue for this?

- [X] There are no existing issues.

### Are the dependencies a supported version?

- [X] The dependencies are supported.

### IBM Z Open Automation Utilities

v1.2.3

### IBM Enterprise Python

v3.9.x

### IBM z/OS Ansible core Version

v1.6.0-beta.1

### ansible-version

v2.14.x

### z/OS version

v2.4

### Ansible module

zos_job_submit

### Bug description

The `zos_job_submit` truncates the final character from its input. See [conversation](https://ibm-systems-z.slack.com/archives/CSLEJ8VGV/p1686652772343239).

Example playbook:

```yaml
---

- name: Running zos_job_submit truncates final character
  hosts: all
  gather_facts: true

  vars:
    jcl_template: "{{ playbook_dir }}/bug.j2"

  tasks:
    - name: Show Ansible version
      ansible.builtin.debug:
        msg: "{{ ansible_version }}"
      delegate_to: localhost

    - name: Show managed node Python version
      ansible.builtin.debug:
        msg: "{{ ansible_python_version }}"

    - name: Get ZOAU version
      environment: "{{ environment_vars }}"
      ansible.builtin.command: zoaversion
      register: zoaversion
      changed_when: false

    - name: Show ZOAU version
      ansible.builtin.debug:
        msg: "{{ zoaversion.stdout }}"

    - name: Show collection version
      environment: "{{ environment_vars }}"
      ansible.builtin.debug:
        msg: "ibm.ibm_zos_core version {{ lookup('community.general.collection_version', 'ibm.ibm_zos_core') }}"

    - name: "Create temp file for job template {{ jcl_template }}"
      ansible.builtin.tempfile:
        state: file
      register: templated_job
      delegate_to: localhost

    - name: "Apply job template {{ jcl_template }}"
      ansible.builtin.template:
        src: "{{ jcl_template }}"
        dest: "{{ templated_job.path }}"
        mode: "600"
      delegate_to: localhost

    - name: "Submit JCL for job {{ jcl_template }}"
      environment: "{{ environment_vars }}"
      ibm.ibm_zos_core.zos_job_submit:
        src: "{{ templated_job.path }}"
        location: LOCAL
      register: job_submit_response

    - name: Set response fact for {{ jcl_template | basename | splitext | first }}
      ansible.builtin.set_fact:
        job_response: "{{ job_submit_response }}"

    - name: "Delete templated job tempfile for {{ jcl_template }}"
      ansible.builtin.file:
        state: absent
        path: "{{ templated_job.path }}"
      delegate_to: localhost
```

Example template:

```jinja2
//TEST JOB REGION=0M,MSGLEVEL=(1,1),MSGCLASS=R,NOTIFY=&SYSUID
//STEP1  EXEC PGM=IEFBR14,MEMLIMIT=0M
//DD1   DD DISP=NEW
```

⚠️  Note no newline at end of file!

### Playbook verbosity output.

```
fatal: [winmvs28]: FAILED! => {
    "changed": false,
    "checksum": "43d9e249317b8a3c049ea074aa77d2f071bcd7e9",
    "dest": "/tmp/ansible.25wsz09m",
    "duration": 0,
    "gid": 5,
    "group": "TSOUSER",
    "invocation": {
        "module_args": {
            "encoding": null,
            "from_encoding": "UTF-8",
            "location": "LOCAL",
            "max_rc": null,
            "return_output": true,
            "src": "/var/folders/8d/yt9452bx0xj3fmzsmkgrj1s00000gn/T/ansible.d2ks3qo7",
            "temp_file": "/tmp/ansible.25wsz09m",
            "to_encoding": "IBM-1047",
            "volume": null,
            "wait": false,
            "wait_time_s": 10
        }
    },
    "jobs": [
        {
            "class": "",
            "content_type": "",
            "ddnames": [
                {
                    "byte_count": "133",
                    "content": [
                        "1                   J E S 2  J O B  L O G  --  S Y S T E M  M V 2 8  --  N O D E  W I N M V S 2 C",
                        "0     ",
                        " 16.17.16 JOB99708 ---- FRIDAY,    16 JUN 2023 ----",
                        " 16.17.16 JOB99708  IRR010I  USERID BENCOX   IS ASSIGNED TO THIS JOB.",
                        " 16.17.16 JOB99708  IEF452I TEST     - JOB NOT RUN - JCL ERROR",
                        " 16.17.16 JOB99708  $HASP396 TEST     TERMINATED",
                        "0------ JES2 JOB STATISTICS ------",
                        "-  16 JUN 2023 JOB EXECUTION DATE",
                        "-            3 CARDS READ",
                        "-           20 SYSOUT PRINT RECORDS",
                        "-            0 SYSOUT PUNCH RECORDS",
                        "-            1 SYSOUT SPOOL KBYTES",
                        "-         0.00 MINUTES EXECUTION TIME"
                    ],
                    "ddname": "JESMSGLG",
                    "id": "2",
                    "proctep": null,
                    "record_count": "13",
                    "stepname": "JES2"
                },
                {
                    "byte_count": "136",
                    "content": [
                        "        1 //TEST JOB REGION=0M,MSGLEVEL=(1,1),MSGCLASS=R,NOTIFY=&SYSUID           JOB99708",
                        "          IEFC653I SUBSTITUTION JCL - REGION=0M,MSGLEVEL=(1,1),MSGCLASS=R,NOTIFY=BENCOX",
                        "        2 //STEP1  EXEC PGM=IEFBR14,MEMLIMIT=0M",
                        "        3 //DD1   DD DISP=NE"
                    ],
                    "ddname": "JESJCL",
                    "id": "3",
                    "proctep": null,
                    "record_count": "4",
                    "stepname": "JES2"
                },
                {
                    "byte_count": "137",
                    "content": [
                        "  STMT NO. MESSAGE",
                        " -    ",
                        "         3 IEF643I UNIDENTIFIED POSITIONAL PARAMETER IN THE DISP FIELD"
                    ],
                    "ddname": "JESYSMSG",
                    "id": "4",
                    "proctep": null,
                    "record_count": "3",
                    "stepname": "JES2"
                }
            ],
            "duration": 1,
            "job_id": "JOB99708",
            "job_name": "TEST",
            "owner": "BENCOX",
            "ret_code": {
                "code": null,
                "msg": "JCL ERROR",
                "msg_code": null,
                "msg_text": "JCLERR",
                "steps": []
            },
            "subsystem": "WINMVS2C",
            "system": "MV28"
        }
    ],
    "md5sum": "2dec3dc1cdc8564254f4563387eb594e",
    "mode": "0600",
    "msg": "The JCL submitted with job id JOB99708 but there was an error, please review the error for further details: The job completion code (CC) was not in the job log. Please review the error JCL ERROR and the job log.",
    "owner": "BENCOX",
    "size": 119,
    "src": "/u/bencox/.ansible/tmp/ansible-tmp-1686928632.747919-71255-57161610142757/source",
    "state": "file",
    "stderr": "The job completion code (CC) was not in the job log. Please review the error JCL ERROR and the job log.",
    "stderr_lines": [
        "The job completion code (CC) was not in the job log. Please review the error JCL ERROR and the job log."
    ],
    "uid": 691
}
```

### Ansible configuration.

```YAML
[defaults]
forks = 25

[ssh_connection]
pipelining = True

[connection]
pipelining = True
```

### Contents of the inventory

```YAML
---

plex2:
  hosts:
    winmvs28:
      ansible_host: "28"
```

### Contents of `group_vars` or `host_vars`

```YAML
---
  
ansible_python_interpreter: "/python/v3r9/usr/lpp/IBM/cyp/v3r9/pyz/bin/python3.9"
ansible_ssh_common_args: "-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"

pyz: "/python/v3r9/usr/lpp/IBM/cyp/v3r9/pyz"
zoau: "/tools/zoautil"

environment_vars:
  _BPXK_AUTOCVT: "ON"
  ZOAU_HOME: "{{ zoau }}"
  PYTHONPATH: "{{ zoau }}/lib"
  LIBPATH: "{{ zoau }}/lib:{{ pyz }}/lib:/lib:/usr/lib:."
  PATH: "{{ zoau }}/bin:{{ pyz }}/bin:/bin:/var/bin"
  _CEE_RUNOPTS: "FILETAG(AUTOCVT,AUTOTAG) POSIX(ON)"
  _TAG_REDIR_ERR: "txt"
  _TAG_REDIR_IN: "txt"
  _TAG_REDIR_OUT: "txt"
  LANG: "C"
```

**Comment by stewartfrancis** [comment]
I'm working around this currently by padding my JCL with a comment statement at the end

**Comment by ddimatos** [comment]
Thank you for opening the issue @ind1go and @stewartfrancis for assisting with troubleshooting. This is supposed to be corrected in ZOAU 1.2.4 (tracker 9884), thus this issue is a validation task for this team and will be put in plan.
